### DISCUSSION TITLE
Issue reading in PDF file

### DISCUSSION THREAD

**Discussion by rutledj** [nextgenhealthcare/connect#5435] [Q&A]
I'm using a javascript reader to read in two files. One is a pipe delimited file that has the patient info, which I convert to a HL7 message. It also contains the name of the associated PDF file that I then try to read in and put in the OBX segment using the below code:

```
//get pdf contents
var folder = "/folders/sftp_users/CTHCare/files/qa/";
//create a directory object
var dir = new java.io.File(folder);
var pdfFile = FileUtil.read(folder + $('pdfFileName'));
tmp['OBX']['OBX.5']['OBX.5.1'] = pdfFile;
```

I get this error:

ER7Serializer error
ERROR MESSAGE: Error converting XML to ER7
org.xml.sax.SAXParseException; lineNumber: 72; columnNumber: 14; Character reference "&#17" is an invalid XML character.

Do I need to encode or serialize the pdf to embed it into the OBX segment?

**Comment by rutledj** [comment]
Just tried using readBytes. It seems to work but I'm getting newlines in the message.

```
var dir = new java.io.File(folder);
var pdfFile = FileUtil.readBytes(folder + $('pdfFileName'));
var encodedPDF = FileUtil.encode(pdfFile);
tmp['OBX']['OBX.5']['OBX.5.3'] = 'PDF';
tmp['OBX']['OBX.5']['OBX.5.4'] = 'PDF'; 
tmp['OBX']['OBX.5']['OBX.5.5'] = encodedPDF;
```

Segment looks like this:

OBX|1|ED|||^^PDF^PDF^JVBERi0xLjUKJdP0zOEKJSBQREZzaGFycCBWZXJzaW9uIDEuNTAuNDAwMC4wICh2ZXJib3NlIG1v
ZGUpCiUgQ3JlYXRpb24gZGF0ZTogMDkvMjEvMjAyMiAwMTowODoyOCAgICAgICAgICAgICAKJSBD
cmVhdGlvbiB0aW1lOiAwLjAxNiBzZWNvbmRzICAgICAgICAgICAgICAgICAgIAolIEZpbGUgc2l6
ZTogMTA0MzY5IGJ5dGVzICAgICAgICAgICAgICAgICAgICAgICAgCiUgUGFnZXM6IDIgICAgICAg
ICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAKJSBPYmplY3RzOiAyNCAgICAgICAgICAg
ICAgICAgICAgICAgICAgICAgICAgICAgIAolLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0t
LS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0t
LS0tLS0tLS0tLS0KMSAwIG9iaiAgICUgUGRmU2hhcnAuUGRmLlBkZkRvY3VtZW50SW5mb3JtYXRp
b24KPDwKICAvQXV0aG9yIDxGRUZGMDA0QjAwNzIwMDY5MDA3MzAwNzQwMDY5MDA2RTAwNjEwMDIw
MDA0ODAwNkYwMDZDMDA2RDAwNjUwMDczPgogIC9DcmVhdGlvbkRhdGUgKEQ6MjAyMjAyMDExNTQ1
MDBaKQogIC9DcmVhdG9yIChDb25nYSBQREYgRW5naW5lIDIwMTUuMDggQSkKICAvTW9kRGF0ZSAo
RDoyMDIyMDkyMTAxMDgyOC0wNicwMCcpCiAgL1Byb2R1Y2VyIChQREZzaGFycCAxLjUwLjQwMDAg
XCh3d3cucGRmc2hhcnAuY29tXCkgXChPcmlnaW5hbDogQ29uZ2EgUERGIEVuZ2luZSAyMDE1LjA4
IEFcKSkKPj4KZW5kb2JqCiUtLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0t
LS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0t
LQoyIDAgb2JqICAgJSBQZGZTaGFycC5QZGYuQWR2YW5jZWQuUGRmQ2F0YWxvZwo8PAogIC9MYW5n
IChlbi1VUykKICAvUGFnZXMgMyAwIFIKICAvVHlwZSAvQ2F0YWxvZwo+PgplbmRvYmoKJS0tLS0t
LS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0t
LS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tCjMgMCBvYmogICAlIFBkZlNoYXJw
LlBkZi5QZGZQYWdlcwo8PAogIC9Db3VudCAyCiAgL0tpZHMgWzQgMCBSIDE4IDAgUl0KICAvVHlw
ZSAvUGFnZXMKPj4KZW5kb2JqCiUtLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0t
LS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0t
LS0tLQo0IDAgb2JqICAgJSBQZGZTaGFycC5QZGYuUGRmUGFnZQo8PAogIC9Db250ZW50cyA1IDAg
UgogIC9Hcm91cAogIDw8CiAgICAvQ1MgL0RldmljZVJHQgogICAgL1MgL1RyYW5zcGFyZW5jeQog
ICAgL1R5cGUgL0dyb3VwCiAgPj4KICAvTWVkaWFCb3ggWzAgMCA2MTIgNzkyXQogIC9QYXJlbnQg
MyAwIFIKICAvUmVzb3VyY2VzCiAgPDwKICAgIC9Gb250CiAgICA8PAogICAgICAvRkFBQUFIIDcg
MCBSCiAgICAgIC9GQUFBQkEgMTAgMCBSCiAgICAgIC9GQUFBQkUgMTQgMCBSCiAgICAgIC9GQUFB

**Reply by pacmano1** [reply]
remove them with a replace.

**Reply by pacmano1** [reply]
https://forums.mirthproject.io/forum/mirth-connect/support/14928-fileutil-encode-adding-line-breaks

sort of covers the entire question you asked here.

**Comment by rutledj** [comment]
Thanks. I've been searching everywhere for an example and didn't find this.  So does the decode process add the carriage returns back in so the pdf displays correctly?
